src/Module/Calendar/Event/Form.php

```diff
@@ -0,0 +1,253 @@
+			throw new BadRequestException($this->t('Invalid Request'));
+		}
+
+		if (!$this->session->getLocalUserId()) {
+			$this->sysMessages->addNotice($this->t('Permission denied.'));
+			return Login::form();
+		}
+
+		$mode = $this->parameters['mode'];
+
+		if (($mode === self::MODE_EDIT || $mode === self::MODE_COPY)) {
+			if (empty($this->parameters['id'])) {
+				throw new BadRequestException('Invalid Request');
+			}
+			$orig_event = EventModel::getByIdAndUid($this->session->getLocalUserId(), $this->parameters['id']);
+			if (empty($orig_event)) {
+				throw new BadRequestException('Invalid Request');
+			}
+		}
+
+		if ($mode === self::MODE_NEW) {
+			$this->session->set('return_path', $this->args->getCommand());
+		}
+
+		// get the translation strings for the calendar
+		$i18n = EventModel::getStrings();
+
+		$this->page->registerStylesheet('view/asset/fullcalendar/dist/fullcalendar.min.css');
+		$this->page->registerStylesheet('view/asset/fullcalendar/dist/fullcalendar.print.min.css', 'print');
+		$this->page->registerFooterScript('view/asset/moment/min/moment-with-locales.min.js');
+		$this->page->registerFooterScript('view/asset/fullcalendar/dist/fullcalendar.min.js');
+
+		$htpl                   = Renderer::getMarkupTemplate('calendar/calendar_head.tpl');
+		$this->page['htmlhead'] .= Renderer::replaceMacros($htpl, [
+			'$calendar_api' => $this->baseUrl . '/calendar/api/get',
+			'$event_api'    => $this->baseUrl . '/calendar/event/show',
+			'$modparams' => 2,
+			'$i18n' => $i18n,
+		]);
+
+		$share_checked  = '';
+		$share_disabled = '';
+
+		if (empty($orig_event)) {
+			$orig_event = User::getById($this->session->getLocalUserId(), ['allow_cid', 'allow_gid', 'deny_cid',
+																		   'deny_gid']);;
+		} else if ($orig_event['allow_cid'] !== '<' . $this->session->getLocalUserId() . '>'
+				   || $orig_event['allow_gid']
+				   || $orig_event['deny_cid']
+				   || $orig_event['deny_gid']) {
+			$share_checked = ' checked="checked" ';
+		}
+
+		// In case of an error the browser is redirected back here, with these parameters filled in with the previous values
+		if (!empty($request['nofinish'])) {
+			$orig_event['nofinish'] = $request['nofinish'];
+		}
+		if (!empty($request['summary'])) {
+			$orig_event['summary'] = $request['summary'];
+		}
+		if (!empty($request['desc'])) {
+			$orig_event['desc'] = $request['desc'];
+		}
+		if (!empty($request['location'])) {
+			$orig_event['location'] = $request['location'];
+		}
+		if (!empty($request['start'])) {
+			$orig_event['start'] = $request['start'];
+		}
+		if (!empty($request['finish'])) {
+			$orig_event['finish'] = $request['finish'];
+		}
+
+		$n_checked = (!empty($orig_event['nofinish']) ? ' checked="checked" ' : '');
+
+		$t_orig = $orig_event['summary'] ?? '';
+		$d_orig = $orig_event['desc'] ?? '';
+		$l_orig = $orig_event['location'] ?? '';
+		$eid    = $orig_event['id'] ?? 0;
+		$cid    = $orig_event['cid'] ?? 0;
+		$uri    = $orig_event['uri'] ?? '';
```

Comment:
Hmm, PHP CS will bark at this.
```suggestion
		$t_orig = $orig_event['summary']  ?? '';
		$d_orig = $orig_event['desc']     ?? '';
		$l_orig = $orig_event['location'] ?? '';
		$eid    = $orig_event['id']       ?? 0;
		$cid    = $orig_event['cid']      ?? 0;
		$uri    = $orig_event['uri']      ?? '';
```